### PR TITLE
Update clamp, clamp_min, clamp_max to use the minimum norm subgradient

### LINKED ISSUE #184572: `torch.maximum` / `torch.clamp_min` gradients at tie point violate PyTorch's documented subgradient convention and are inconsistent with `F.relu`
### 🐛 Describe the bug

`torch.maximum(x, tensor(0))`, `torch.clamp_min(x, 0)`, and `torch.clamp(x, min=0)` all implement `max(x, 0)`, but their gradients at `x=0` contradict PyTorch's own autograd documentation and are mutually inconsistent.

## Minimal reproducer

```python
import torch
import torch.nn.functional as F

formulations = [
    ("F.relu(x)",                   lambda x: F.relu(x)),
    ("torch.maximum(x, tensor(0))", lambda x: torch.maximum(x, torch.tensor(0.0))),
    ("torch.clamp_min(x, 0)",       lambda x: torch.clamp_min(x, 0.0)),
    ("torch.clamp(x, min=0)",       lambda x: torch.clamp(x, min=0.0)),
]

for name, fn in formulations:
    x = torch.tensor(0.0, requires_grad=True)
    fn(x).backward()
    print(f"{name:<35} grad at x=0: {x.grad.item()}")
```

## Observed output

```
F.relu(x)                           grad at x=0: 0.0
torch.maximum(x, tensor(0))         grad at x=0: 0.5
torch.clamp_min(x, 0)               grad at x=0: 1.0
torch.clamp(x, min=0)               grad at x=0: 1.0
```

## Expected output

```
F.relu(x)                           grad at x=0: 0.0
torch.maximum(x, tensor(0))         grad at x=0: 0.0
torch.clamp_min(x, 0)               grad at x=0: 0.0
torch.clamp(x, min=0)               grad at x=0: 0.0
```

## Why 0.0 is the correct answer

All four expressions compute the same mathematical function `max(x, 0)`, yet they produce three different gradients at `x=0`. This is a problem in practice: users who refactor between these equivalent formulations silently get different gradient behaviour at zero-initialised weights or ReLU activations that land exactly on zero.

PyTorch's own [autograd documentation](https://pytorch.org/docs/stable/notes/autograd.html) specifies:

> *"If the function is convex (at least locally), use the sub-gradient of minimum norm."*

`max(x, 0)` is convex. Its subdifferential at `x=0` is the interval `[0, 1]`. The element
of minimum norm is **0.0**.

| Formulation | grad at x=0 | Consistent with docs? |
|---|---|---|
| `F.relu(x)` | 0.0 | ✓ |
| `torch.abs(x)` | 0.0 | ✓ |
| `torch.maximum(x, tensor(0))` | 0.5 | ✗ |
| `torch.clamp_min(x, 0)` | 1.0 | ✗ |
| `torch.clamp(x, min=0)` | 1.0 | ✗ |

`F.relu` and `torch.abs` already follow the documented minimum-norm convention. `torch.maximum` and `torch.clamp` do not.

### Versions

```
PyTorch version: 2.13.0.dev20260512+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: 18.1.3 (1ubuntu1)
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.12.3 (main, Mar 23 2026, 19:04:32) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.14.0-37-generic-x86_64-with-glibc2.39
Is CUDA available: True
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 5090
Nvidia driver version: 590.48.01

[pip3] numpy==2.4.4
[pip3] torch==2.13.0.dev20260512+cu130
[pip3] triton==3.7.0+git88b227e2
```

cc @ezyang @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93

### PR DESCRIPTION
**Pull request by nrspo**
Fixes #184572 

`torch.clamp`, `torch.clamp_min`, `torch.clamp_max` subgradients violate PyTorch's documented minimum norm convention, returning 1.0 instead of 0.0. 

Changes made:
1. `tools/autograd/derivatives.yaml`: Update boundary conditions for clamp_min and clamp_max
2. `torch/csrc/autograd/FunctionsManual.cpp`: Update boundary conditions in `clamp_backward`
3. `test/test_autograd.py`: Create class `TestAutogradClampGradient(TestCase)` to test boundary conditions output 0.

### CODE REVIEW

**Comment by Skylion007** [comment]
Superceded by: https://github.com/pytorch/pytorch/pull/191142
